### ISSUE TITLE
`auracle download` fails silently for valid package name

### ISSUE THREAD

**Issue by mrkline** [falconindy/auracle#44]
When trying to grab [linux-ck](https://aur.archlinux.org/packages/linux-ck/), `auracle download linux-ck` immediately exits with error code 1, without printing any error.

I haven't had too much time to debug myself, but I was able to deduce that `Auracle::Download()` is hitting the `return -ENOENT` [at the bottom of the function](https://github.com/falconindy/auracle/blob/4a94781e696326e9b0ea4362666376c7a8e8bdc9/src/auracle/auracle.cc#L477).

`auracle info linux-ck` and friends all work fine, so something's amiss.

**Comment by falconindy** [comment]
Works for me. I don't know how you've determined that the ENOENT you've found is exactly the error that auracle trips on.

```
$ auracle download linux-ck
download complete: /home/noclaf/tmp/linux-ck
```

**Comment by mrkline** [comment]
I built from source and did some poking around. On the current `master` (4a94781e696326e9b0ea4362666376c7a8e8bdc9):

```diff
diff --git a/src/auracle/auracle.cc b/src/auracle/auracle.cc
index 1386be5..23125fb 100644
--- a/src/auracle/auracle.cc
+++ b/src/auracle/auracle.cc
@@ -3,6 +3,7 @@
 #include <archive.h>
 #include <archive_entry.h>
 
+#include <cstdio>
 #include <cerrno>
 #include <filesystem>
 #include <functional>
@@ -474,6 +475,7 @@ int Auracle::Download(const std::vector<std::string>& args,
   }
 
   if (iter.package_cache.empty()) {
+    printf("Hmm...\n");
     return -ENOENT;
   }
```

gives

```
$ ./build/auracle download linux-ck
Hmm...
$ echo $?
1
```

I've seen this on several of my machines now - do you think it could be config related?

**Comment by falconindy** [comment]
I'd guess you have linux-ck mentioned in IgnorePkg in /etc/pacman.conf.

**Comment by mrkline** [comment]
Ah, the issue was because I had `linux-ck` added to my `IgnorePkg` in `pacman.conf`. I'd like to keep it there - I use a different AUR helper for other packages, but the caching it does doesn't play well with building kernels, which is why I've been using `cower` and now `auracle` for them.

Two questions:

1. Would you be opposed to a patch that adds some "Skipping <arg>, ignored in pacman.conf" warning when this happens?

2. Is there a straightforward way to make auracle ignore the config? It seems like I can point the undocumented `pacmanconfig` flag at `/dev/null`, but that seems kludgy.

**Comment by falconindy** [comment]
I'm inclined to rip out all of the IgnorePkg handling at this point.

**Comment by falconindy** [comment]
No longer an issue.

**Comment by bbo2adwuff** [comment]
> I'm inclined to rip out all of the IgnorePkg handling at this point.

I was using IgnorePkg, but now I'm wondering how to ignore packages when using `auracle download -r 'auracle sync -q'`

**Comment by falconindy** [comment]
Without understanding *why* you want this, I can't possibly help. It's unlikely that IgnorePkg handling will come back in any form.

**Comment by bbo2adwuff** [comment]
Well, I use a script for updates. That script calls pacman, auracle, etc. I chose auracle to keep my AUR packages up to date. I used `auracle download -r 'auracle sync -q'` to update AUR packages.

For university, I rely on an older version of a AUR package for which I used `IgnorePkg` so far.

Well my work around for now is to use `sed` to remove the package the I want to ignore before passing `auracle sync -q` to `auracle download`.
